avocado/plugins/beaker_result.py

```diff
@@ -0,0 +1,137 @@
+"""
+beaker result plugin
+
+Sends results and logs to beaker using the harness API.
+https://beaker-project.org/docs/alternative-harnesses/index.html
+
+Finds the beaker API entry point using the BEAKER_LAB_CONTROLLER_URL
+environment variable.  Does nothing in case the variable is not set.
+
+"""
+import glob
+import os
+import pprint
+import urllib.error
+import urllib.parse
+import urllib.request
+
+from avocado.core.output import LOG_UI
+from avocado.core.plugin_interfaces import ResultEvents
+
+
+class BeakerResult(ResultEvents):
+
+    """
+    send test results to beaker test harness api
+    """
+
+    name = 'beaker'
+    description = 'report results to beaker'
+
+    beaker_url = None
+    job_id = None
+
+    def __init__(self, config):
+        self.config = config  # make lint happy
+        baseurl = os.environ.get('BEAKER_LAB_CONTROLLER_URL')
+        recipeid = os.environ.get('RSTRNT_RECIPEID')
+        taskid = os.environ.get('RSTRNT_TASKID')
+        if baseurl is None or recipeid is None or taskid is None:
+            return
+        baseurl = baseurl.rstrip('/')
+        self.beaker_url = baseurl + '/recipes/' + recipeid + '/tasks/' + taskid
+        LOG_UI.info("beaker: using API at %s (R:%s T:%s)", baseurl, recipeid, taskid)
```

Comment:
The HTML result plugin does use the `LOG_UI`, so it's a clear precedent.  I wonder, though, if this is level of verbosity is desirable.  For instance, I see on a simple run:

```
beaker: using API at https://foo/bar (R:100 T:101)
JOB ID     : bffdb1fc9bff67ab3a5311d0eafb47b83509ada9
JOB LOG    : /home/cleber/avocado/job-results/job-2021-11-22T11.13-bffdb1f/job.log
 (1/1) /bin/true: STARTED
beaker: POST https://foo/bar/recipes/100/tasks/101/results/ ...
 (1/1) /bin/true: PASS (0.01 s)
beaker: PUT https://foo/bar/recipes/100/tasks/101/logs/bffdb1-results.tap ...
beaker: PUT https://foo/bar/recipes/100/tasks/101/logs/bffdb1-job.log ...
beaker: PUT https://foo/bar/recipes/100/tasks/101/logs/bffdb1-id ...
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/cleber/avocado/job-results/job-2021-11-22T11.13-bffdb1f/results.html
JOB TIME   : 15.94 s
```

I think it'd be more inline with the current UI to simply output one line at the end, pointing to the base location where the logs can be accessed, similar to the `JOB HTML` line.

Reply:
Is there some way to keep the line but not print it by default?  Tried to use 'debug' for the request logging (the PUT lines), didn't work ...

Reply:
> 

There should be... but I'm having trouble with that.  We've had a lot of changes to the logging handling lately, so there may be something funky going on.  I'll check it out and give you a proposal here.